### PR TITLE
Fix tests for windows, and run CI on windows

### PR DESCRIPTION
**Pull request by prashantv**

### CODE REVIEW

File: cmd/errtrace/main_test.go [hunk 1]
```diff
@@ -391,16 +391,18 @@ func extractLogs(src []byte) ([]logLine, error) {
...
 		if line == "" {
 			continue
 		}
 
+		// Drop the path so we can determinstically split on ":" (which is a valid character in Windows paths).
+		line = strings.TrimPrefix(line, file)
 		parts := strings.SplitN(line, ":", 4)
 		if len(parts) != 4 {
-			return nil, fmt.Errorf("bad log line: %q", line)
+			return nil, fmt.Errorf("bad log line1: %q", line)
```
Comment (abhinav) [inline cmd/errtrace/main_test.go:405]: "line1"?

File: cmd/errtrace/main_test.go [hunk 2]
```diff
@@ -411,13 +413,13 @@ func parseLogOutput(s string) ([]logLine, error) {
 			msg = parts[3] // file:line:column:msg
 		}
 
-		line, err := strconv.Atoi(parts[1])
```
Comment (abhinav) [inline cmd/errtrace/main_test.go:414]: lol derp

File: internal/tracetest/clean.go [hunk 3]
```diff
@@ -52,7 +52,11 @@ func MustClean(trace string) string {
 func getErrtraceDir() string {
 	_, file, _, _ := runtime.Caller(0)
 	// Note: Assumes specific location of this file in errtrace, strip internal/tracetest/<file>
-	return filepath.Dir(filepath.Dir(filepath.Dir(file)))
+	dir := filepath.Dir(filepath.Dir(filepath.Dir(file)))
+
+	// On Windows, filepath.Dir cleans the path, which modifies the separator.
+	// To get back the original separator, truncate the original string.
```
Comment (abhinav) [inline internal/tracetest/clean.go:58]: 🤣

**Review by abhinav — APPROVED** [review]

**Comment by abhinav** [comment]
For the error, I think you need to set shell = bash

**Comment by abhinav** [comment]
e.g. I do this elsewhere:

https://github.com/abhinav/doc2go/blob/d60ec80e38f0195fc7ef34ce2be377941bf6c222/Makefile#L59-L60

https://github.com/abhinav/doc2go/blob/d60ec80e38f0195fc7ef34ce2be377941bf6c222/.github/workflows/ci.yml#L67-L69

File: cmd/errtrace/main_test.go (hunk 1 above)
Comment (prashantv) [reply cmd/errtrace/main_test.go]: removed, debugging leftover (to identify which "bad log line" print was causing the issue)

**Pull request merged**